x-pack/legacy/plugins/monitoring/public/views/base_controller.js

```diff
@@ -110,6 +128,28 @@ export class MonitoringViewBaseController {
       timefilter.enableAutoRefreshSelector();
     }
 
+    const updateHandler = () => {
+      const { from, to } = timefilter.getTime();
+      const parsedRange = {
+        from: dateMath.parse(from),
+        to: dateMath.parse(to),
+      };
+
+      if (invalidTimeRange(parsedRange)) {
+        if (to === 'now') {
+          timefilter.setTime(VALID_RANGE);
+        } else {
+          const { from } = parsedRange;
+          const range = { from, to: from.clone().add(MAX_RANGE_SECONDS, 'seconds'), mode: 'absolute' };
```

Comment:
I'm not sure we want this behavior either. Ideally, if I try and set the time range to a window of time too small, it either reverts back to the window I had previously, or maybe even better, it just leaves the time as-is but shows the toast over and over until the user changes the time window. I don't think we should be changing the mode on the user since they might be in something other than `absolute` once they encounter this toast

Reply:
I could keep state of their current valid range and use that instead.

> I don't think we should be changing the mode on the user since they might be in something other than absolute once they encounter this toast

Relative time is already handled in the if clause itself

Reply:
I could be off here since I'm not super familiar with Kibana code any more but this sounds like a use case for adding a validation hook to the time picker itself, before it decides to change the time bounds. Then we (Monitoring code) could use this hook to perform the necessary validation and reject it if the window is too small.

Obviously this would involve changes to the time picker code itself but if this feels like the right solution, I think it might be worth putting up a separate PR to add the validation hook and see what the Kibana team thinks of it.

Reply:
@igoristic I don't see that in the code and I'm not seeing that behavior:
![out](https://user-images.githubusercontent.com/56682/61662098-a5053100-ac9b-11e9-9034-1d19f4b171e5.gif)

Reply:
@ycombinator Yeah, I agree the date picker should have that functionality/feature, or at least something like `e.stopImmediatePropagation()` (so we can hijack and override at will). 

It does kinda feel like a bandaid fix for the underline issue. Which makes me think, what is the true source of the bug? Maybe this is something that should be fixed at a lower level instead? Either in our backend server API route's response, or maybe even in Elasticsearch itself?

@chrisronline You're right! I have the Relative vs Absolute covered in the little clock dropdown icon, but not in the additional range selectors. Don't know why there are multiple entry-points, but this makes it kind of confusing. I will add this fix to both 👍

Reply:
> Maybe this is something that should be fixed at a lower level instead? Either in our backend server API route's response, or maybe even in Elasticsearch itself?

This is a very good point. I'm +1 for looking into whether we can catch this specific bad scenario in our server-side code and make it return a better error message than what's shown in the toast right now. Then we wouldn't need to worry about catching it at two places in the browser-side code (timepicker and chart brush).

Reply:
Hey @igoristic, any updates here?